models/viralspread/netlogo/TryRodrigo.nlogo

```diff
@@ -0,0 +1,664 @@
+;; Create "breed" of 'turtle' called cell
+breed [cells cell]          ;; eventually separate mesophyll?
+breed [vasculars vascular]  ;; phloem vascular bundles 
+
+;; Create "breed" of 'link' called phloem
+directed-link-breed [phloems phloem]  
+;; Note: I know there shouldn't be an s, but it wanted a different plural :(
+
+;; Declare global variables
+globals
+[
+  num-infected   ;; keep track of number infected
+]
+
+;; Declare the cell-specific (turtle breed) variables
+cells-own
+[
+  infected?         ;; if true, the cell is infectious
+  resistant?        ;; if true, cell can't be infected
+  num-plasmodesmata ;; number of connections with other cells
+]
+
+
+
+;; Setup procedures
+to setup
+  clear-all    ;; remove anything from previous runs
+  setup-cells  ;; set up the cells
+  setup-leaf   ;; set up the connections between cells
+  ask n-of multiplicity-of-infection cells   ;; determine how many viruses
+    [ become-infected ]
+  ask links [ set color white ]  ;; make the symplastic connections white
+  ask phloems 
+    [ 
+      set color green      ;; make the vascular bundles green and thicker
+      set thickness 3
+    ]
+  reset-ticks  ;; reset timer from previous run
+end
+
+to setup-cells
+  set-default-shape cells "circle"   ;; uses circle shape for displayed cells
+  create-cells num-cells
+  [
+    setxy (random-xcor * 0.95) (random-ycor * 0.95) ;; no nodes too close to edges
+    become-susceptible
+  ]
+end
+
+to setup-leaf
+  ask cells 
+  [ 
+    while [num-plasmodesmata < avg-num-plasmodesmata]
+    [ 
+      let choice (min-one-of (other cells with [not link-neighbor? myself])
+                              [distance myself])
+      if choice != nobody [ create-link-with choice ]
+      set num-plasmodesmata num-plasmodesmata + 1
+    ]
+  ]
+  repeat 10 ; to make the network look nicer
+  [ layout-spring turtles links 0.3 (world-width / (sqrt num-cells)) 1 ]
+end
+
+
+
+to go
+  ;; once all cells are infected, run 40 more times, then stop
+;;  if all? cells [ infected? ] 
+;;  [  
+;;    let i 0 
+;;    while [i <= 40] 
+;;    [
+;;      spread-virus
+;;      set i i + 1
+;;      tick
+;;    ]
+;;    stop  ;; stop model completely
+;;  ]
+  if ticks > num-ticks [ stop ]
+  ;; otherwise, continue to spread the virus
+  spread-virus
+  tick
+end
+
+
+
+;; Cell Procedures
+to become-infected
+  set infected? true
+  set resistant? false
+  set color violet
+end
+
+to become-susceptible
+  set infected? false
+  set resistant? false
+  set color green
+end
+
+to become-resistant
+  set infected? false
+  set resistant? true
+  set color blue
+  ask my-links [ set color blue - 2 ] ;; make neighbours resistant as well
+end
+
+
+;; Virus Procedure
+to spread-virus
+  ask cells with [infected?]
+    [ 
+      ask link-neighbors with [not resistant?]
+        [ 
+          if (random-float 100 < viral-spread-chance) and (not infected?)
```

Comment:
So basically, I think we now want some function like "viral-spread-from-neighbours" based on a cells-own "num-virions" paramter, rather than a random-float?